openingd/dualopend.c

```diff
@@ -352,100 +359,216 @@ static bool find_txout(struct wally_psbt *psbt, const u8 *wscript, u16 *funding_
 	return false;
 }
 
-static bool check_balances(struct state *state,
+static bool check_balances(const tal_t *ctx,
+			   struct state *state,
 			   struct wally_psbt *psbt,
-			   bool check_opener_balance,
-			   bool do_full_tx_check,
-			   u32 feerate_per_kw_funding)
+			   u32 feerate_per_kw_funding,
+			   char **err_reason)
```

Comment:
Better interface would be to return char *, and NULL on success.

Reply:
good point!